include/range/v3/view/remove_when.hpp

```diff
@@ -0,0 +1,181 @@
+      , private box<semiregular_t<Fun>>
+    {
+        remove_when_view() = default;
+        constexpr remove_when_view(Rng rng, Fun fun)
+          : remove_when_view::view_adaptor{detail::move(rng)}
+          , remove_when_view::box(detail::move(fun))
+        {}
+
+    private:
+        friend range_access;
+        bool zero_;
+
+        struct adaptor : adaptor_base
+        {
+            adaptor() = default;
+            constexpr adaptor(remove_when_view & rng) noexcept
+              : rng_(&rng)
+            {}
+            static constexpr iterator_t<Rng> begin(remove_when_view & rng)
+            {
+                return *rng.begin_;
+            }
+            constexpr void next(iterator_t<Rng> & it) const
+            {
+                RANGES_ASSERT(it != ranges::end(rng_->base()));
+                rng_->satisfy_forward(++it);
+            }
+            void advance() = delete;
+            void distance_to() = delete;
+
+        private:
+            remove_when_view * rng_;
+        };
+        constexpr adaptor begin_adaptor()
+        {
+            cache_begin();
+            return {*this};
+        }
+        constexpr adaptor end_adaptor()
+        {
+            return {*this};
+        }
+
+        constexpr void satisfy_forward(iterator_t<Rng> & it)
+        {
+            if(zero_)
+            {
+                // if the last match consumed zero elements, we already bumped the
+                // position so can stop here.
+                zero_ = false;
+                return;
+            }
+
+            auto const last = ranges::end(this->base());
+            auto & fun = this->remove_when_view::box::get();
+            if(it != last)
+            {
+                auto p = invoke(fun, it, last);
+                if(p.first)
+                {
+                    zero_ = (it == p.second);
+                    it = p.second;
+                }
+            }
+        }
+
+        constexpr void cache_begin()
+        {
+            if(begin_)
+                return;
+            auto it = ranges::begin(this->base());
+            zero_ = false;
+            satisfy_forward(it);
+            begin_.emplace(std::move(it));
+        }
+
+        detail::non_propagating_cache<iterator_t<Rng>> begin_;
+    };
+
+#if RANGES_CXX_DEDUCTION_GUIDES >= RANGES_CXX_DEDUCTION_GUIDES_17
+    CPP_template(typename Rng, typename Fun)(requires CopyConstructible<Fun>)
+        remove_when_view(Rng &&, Fun)
+            ->remove_when_view<view::all_t<Rng>, Fun>;
+#endif
+
+    namespace view
+    {
+        struct remove_when_fn
+        {
+        private:
+            friend view_access;
+            template<typename Fun>
+            static auto bind(remove_when_fn remove_when, Fun && fun)
+            {
+                return make_pipeable(std::bind(
+                    remove_when, std::placeholders::_1, static_cast<Fun &&>(fun)));
```

Comment:
Please rebase on master and replace `std::bind` with `bind_back`, and make this function `constexpr`. (You should also `#include <range/v3/functional/bind_back.hpp>`.)